framework/web/filter/cachestrategy.go

```diff
@@ -17,18 +18,26 @@ type (
 )
 
 //Filter - implements flamingo filter interface
-func (f *CacheStrategy) Filter(ctx context.Context, r *web.Request, w http.ResponseWriter, chain *web.FilterChain) web.Result {
+func (f *cacheStrategy) Filter(ctx context.Context, r *web.Request, w http.ResponseWriter, chain *web.FilterChain) web.Result {
 	response := chain.Next(ctx, r, w)
-	if defaultResponse, ok := response.(*web.Response); ok {
-		if defaultResponse.CacheDirectives == nil && r.Request().Method == http.MethodGet {
-			cacheStrategy := web.CacheStrategy{}
-			cacheStrategy.SetIsReusable(f.DefaultIsReuseable)
-			cacheStrategy.SetRevalidateEachTime(f.DefaultRevalidateEachTime)
-			cacheStrategy.SetAllowIntermediateCaches(f.DefaultAllowIntermediateCaches)
-			cacheStrategy.SetMaxCacheLifetime(int(f.DefaultMaxCacheLifetime))
-			defaultResponse.CacheDirectives = cacheStrategy.Build()
-			return defaultResponse
-		}
+	if r.Request().Method != http.MethodGet {
+		return response
+	}
+	switch typedResponse := response.(type) {
+	case *web.RenderResponse:
```

Comment:
Is this supposed to be `*web.RenderResponse`? You might want to use `*web.Response`?